paddle/framework/multigpu.md

```diff
@@ -0,0 +1,64 @@
+# Design Doc: Multi-GPU support in Operation Graph
+
+## Abstract
+
+This Design Doc refers to the multi-GPU feature in  paddle.  We propose an approach to support multi-GPU both on a single machine and multiple machines. Every device only run sub-graphs which our framework issued. We use `Broadcast`, `Allreduce` operators to join different device sub-graph to the whole graph.
+
+
+
+## Motivation
+
+Paddle supports training with multiple CPUs and GPUs, refer to different physical devices. We need to support multi-GPU training in parallel for acceleration, in detail, there are two aspects. 
+
+- GPU Data Parallelism 
+
+  Suppose to we have `n`GPUs, every GPU has `1/n`part of training data, and store a complete model in GPU memory.  
+
+- GPU Model Parallelism
+
+  every GPU have part of a complete model in GPU memory.
+
+At the beginning of training, the framework needs to issue the same sub-graph to every GPU in Data Parallelism, or different sub-graph in Model Parallelism.
+
+During training, we need the operations of peer to peer copy between different GPUs, aggregating gradients/parameters from GPUs, and broadcasting parameters to GPUs. Every GPU only need to run the sub-graph with correct place information.
+
+Besides, it needs interfaces to synchronize model update with each other, and issue/merge model from different GPU Cards. 
+
+## Implementation
+
+As mentioned above, we summarise that several kinds of operators are needed. Currently, we need to issue parameters to different GPUs,  named it with Broadcast operator.  And also synchronize parameters between GPUs, called it with AllReduce. 
+
+### Graph Converter
+
+To be compatible with parameter server design doc, the graph converter converts the user defined operation graph into sub-graphs to be executed on different devices.
+
+1. The user-defined operator graph will be partitioned into sub-graph. 
+
+2. Control operators between GPUs will be inserted into the graph.
+
+   *Broadcast, AllReduce in a single machine. And Broadcast, AllReduce, Send, Recv in multiple machines*
```

Comment:
Since you mentioned "Send, Recv" can you please add a reference link to these design docs?